### PR TITLE
Add 'infile' option

### PR DESCRIPTION
**Pull request by binomialstew**
As discussed [here](https://github.com/TriPSs/conventional-changelog-action/issues/219). I still need to test this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ This action will bump version, tag commit and generate a changelog with conventi
 - **Optional** `git-branch`: The branch used to push. Default is the current branch (`${{ github.ref }}`)
 - **Optional** `git-url`: Git repository domain. Default is `github.com`
 - **Optional** `git-path`: Path filter for the logs. If set, only commits that match the path filter will be considered. By default, we won't use this feature(empty string).
+- **Optional** `infile`: Read the CHANGELOG from this file.
```
Comment (TriPSs) [inline README.md]: Could we bring this one inline with `output-file`? So `input-file`, maybe also describe a bit more what it does? If I understood correct :`Read the CHANGELOG from this file and prepend the new release to it`.

File: action.yml [hunk 2]
```diff
@@ -56,6 +56,11 @@ inputs:
     default: "v"
     required: false
 
+  infile:
+    description: "Read the CHANGELOG from this file"
+    default: "CHANGELOG.md"
```
Comment (TriPSs) [inline action.yml]: Also wondering what happens here? Wil this now not cause the whole old changelog to be append to the new one? (if you generate more then only the latest release, 5 is currently the default)

Think it's better to not have a default here?

**Comment by binomialstew** [comment]
@TriPSs, I've addressed your comments

File: action.yml (hunk 2 above)
Comment (binomialstew) [reply action.yml]: Good point. In my tests, it retains the prior content of the `CHANGELOG.md` as long as it is set as `--infile`—so all prior releases. This means that `release-count` will not consider any releases in `input-file`. I will add a note about this to `release-count`

Comment (binomialstew) [reply action.yml]: OK updated

**Comment by binomialstew** [comment]
@TriPSs, it looks like the infile option [is exclusive to command line usage](https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog-cli/cli.js#L123). I'm testing some things now and will update this when possible.

**Comment by binomialstew** [comment]
OK. This has been updated, and I tested the action in my own project to verify:

* That all releases are maintained (tested 7 releases) when using `input-file` option
* That only the default number of releases are maintained (5) when neither `input-file` or `release-count` are used
* If `release-count` is set, the releases are limited to that number

**Comment by TriPSs** [comment]
Thanks for the PR! Could you also add a test case for this?

**Comment by binomialstew** [comment]
> Thanks for the PR! Could you also add a test case for this?

OK. Test case is added. Let me know if this is in line with what you were thinking.

**Comment by TriPSs** [comment]
Thanks for the PR!

**Comment by binomialstew** [comment]
Thanks for merging!

**Pull request merged**